app/models/child_theme.rb

```diff
@@ -1,6 +1,24 @@
 class ChildTheme < ActiveRecord::Base
   belongs_to :parent_theme, class_name: 'Theme'
   belongs_to :child_theme, class_name: 'Theme'
+
+  validate :child_validations
+
+  private
+
+  def child_validations
+    if ChildTheme.exists?(["parent_theme_id = ? OR child_theme_id = ?", child_theme_id, parent_theme_id])
+      errors.add(:base, I18n.t("themes.errors.no_multilevels_components"))
+    end
+
+    if Theme.exists?(id: child_theme_id, user_selectable: true)
```

Comment:
Same as https://github.com/discourse/discourse/pull/6199/files#r207779563

Reply:
I'm not sure I understand here. What I want to do here is make sure the child theme is not user selectable. How do I do that with uniqueness validation?

Reply:
Oops ignore me here